src/ILCompiler.WebAssembly/src/Compiler/WebAssemblyMetadataBlockingPolicy.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Internal.TypeSystem;
+
+namespace ILCompiler.Compiler
+{
+    public class WebAssemblyMetadataBlockingPolicy: MetadataBlockingPolicy
+    {
+        public override bool IsBlocked(MetadataType type)
+        {
+            return false;
```

Comment:
Changing this to `return !(type is EcmaType);` should get rid of some of the other problematic code I pointed to.